projects/storefrontlib/src/layout/theme/site-theme.service.ts

```diff
@@ -0,0 +1,38 @@
+import {
+  ComponentRef,
+  Injectable,
+  Renderer2,
+  RendererFactory2,
+} from '@angular/core';
+import { SiteContextConfig, THEME_CONTEXT_ID } from '@spartacus/core';
+
+@Injectable({ providedIn: 'root' })
+export class SiteThemeService {
+  protected rootComponent: ComponentRef<any>;
+  protected renderer: Renderer2;
+  protected existingTheme: string;
+
+  constructor(
+    protected config: SiteContextConfig,
+    protected rendererFactory: RendererFactory2
+  ) {}
+
+  init(rootComponent: ComponentRef<any>): void {
+    this.renderer = this.rendererFactory.createRenderer(null, null);
+    this.rootComponent = rootComponent;
+    this.setTheme(this.config.context[THEME_CONTEXT_ID]?.[0]);
```

Comment:
can we add an inline doc to explain why we take the first theme only? Just to help devs understand that we're mixing the backend singular value into a more generic multi value property.